### PR TITLE
Fix/snapshot cleanup partitions

### PR DESCRIPTION
**Pull request by knazir**
Issue: https://issues.cask.co/browse/HYDRATOR-578
Build: http://builds.cask.co/browse/HYP-BAD127-1

### CODE REVIEW

File: core-plugins/src/main/java/co/cask/hydrator/plugin/dataset/SnapshotFileSet.java [hunk 1]
```diff
@@ -52,6 +54,7 @@
   private static final Type MAP_TYPE = new TypeToken<Map<String, String>>() { }.getType();
   private static final Gson GSON = new Gson();
   private static final String STATE_FILE_NAME = "state";
+  private static final Logger LOG = LoggerFactory.getLogger(SnapshotFileSet.class);
```
Comment (anwar6953) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/dataset/SnapshotFileSet.java]: unused

File: core-plugins/src/test/java/co/cask/hydrator/plugin/common/SnapshotFileSetTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    PartitionKey key = PartitionKey.builder().addLongField("snapshot", System.currentTimeMillis() - 1000 * 60).build();
+    PartitionOutput partition = partitionedFileSet.getPartitionOutput(key);
+    partition.addPartition();
+
+    SnapshotFileSet snapshotFileSet = new SnapshotFileSet(partitionedFileSet);
+    snapshotFileSet.deleteMatchingPartitionsByTime(System.currentTimeMillis());
+    Assert.assertNull(partitionedFileSet.getPartition(key));
+  }
+
```
Comment (knazir) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/common/SnapshotFileSetTest.java:63]: remove extra newline

File: core-plugins/src/main/java/co/cask/hydrator/plugin/dataset/SnapshotFileSet.java (hunk 1 above)
Comment (knazir) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/dataset/SnapshotFileSet.java]: unused

File: core-plugins/src/test/java/co/cask/hydrator/plugin/common/SnapshotFileSetTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  public void testCleanup() throws Exception {
+    addDatasetInstance(NamespaceId.DEFAULT.toId(),
+                         PartitionedFileSet.class.getName(), "pfs", PartitionedFileSetProperties.builder()
+                            .setPartitioning(Partitioning.builder().addLongField("snapshot").build())
+                            .setInputFormat(AvroKeyInputFormat.class)
+                            .setOutputFormat(AvroKeyOutputFormat.class)
+                            .build());
+    PartitionedFileSet partitionedFileSet = (PartitionedFileSet) getDataset("pfs").get();
+
+    PartitionKey key = PartitionKey.builder().addLongField("snapshot", System.currentTimeMillis() - 1000 * 60).build();
```
Comment (anwar6953) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/common/SnapshotFileSetTest.java]: add a partition with a later time, that doesn't get deleted.

File: core-plugins/src/main/java/co/cask/hydrator/plugin/dataset/SnapshotFileSet.java [hunk 4]
```diff
@@ -139,7 +142,7 @@ public void onSuccess(long snapshotTime) throws IOException, InterruptedExceptio
 
   public void deleteMatchingPartitionsByTime(long upperLimit) throws IOException {
     if (upperLimit > 0 && upperLimit < Long.MAX_VALUE) {
-      PartitionFilter filter = PartitionFilter.builder().addRangeCondition("millisecond", null, upperLimit).build();
+      PartitionFilter filter = PartitionFilter.builder().addRangeCondition("snapshot", null, upperLimit).build();
```
Comment (anwar6953) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/dataset/SnapshotFileSet.java]: put the field name in a constant?

File: core-plugins/src/test/java/co/cask/hydrator/plugin/common/SnapshotFileSetTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import org.junit.Test;
+
+/**
+ * Test class for {@link SnapshotFileBatchSink}.
+ */
+public class SnapshotFileSetTest extends HydratorTestBase {
+  @Test
+  public void testCleanup() throws Exception {
+    addDatasetInstance(NamespaceId.DEFAULT.toId(),
+                         PartitionedFileSet.class.getName(), "pfs", PartitionedFileSetProperties.builder()
```
Comment (sagarkapare) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/common/SnapshotFileSetTest.java]: nit: indentation

**Comment by knazir** [comment]
Build running: http://builds.cask.co/browse/HYP-BAD127-1

**Comment by sagarkapare** [comment]
LGTM 👍

**Comment by knazir** [comment]
Guild is breen, going to merge.

**Pull request merged**